mnefun/_mnefun.py

```diff
@@ -2123,4 +2135,72 @@ def _headpos(p, file_in, file_out):
-    return pos
\ No newline at end of file
+    return pos
+
+
+def info_sss_basis(info, origin='auto', int_order=8, ext_order=3,
+                   coord_frame='head', regularize='in', ignore_ref=True):
+    """Compute the SSS basis for a given measurement info structure
+
+    Parameters
+    ----------
+    info : instance of io.Info
+        The measurement info.
+    origin : array-like, shape (3,) | str
+        Origin of internal and external multipolar moment space in meters.
+        The default is ``'auto'``, which means a head-digitization-based
+        origin fit when ``coord_frame='head'``, and ``(0., 0., 0.)`` when
+        ``coord_frame='meg'``.
+    int_order : int
+        Order of internal component of spherical expansion.
+    ext_order : int
+        Order of external component of spherical expansion.
+    coord_frame : str
+        The coordinate frame that the ``origin`` is specified in, either
+        ``'meg'`` or ``'head'``. For empty-room recordings that do not have
+        a head<->meg transform ``info['dev_head_t']``, the MEG coordinate
+        frame should be used.
+    destination : str | array-like, shape (3,) | None
+        The destination location for the head. Can be ``None``, which
+        will not change the head position, or a string path to a FIF file
+        containing a MEG device<->head transformation, or a 3-element array
+        giving the coordinates to translate to (with no rotations).
+        For example, ``destination=(0, 0, 0.04)`` would translate the bases
+        as ``--trans default`` would in MaxFilter™ (i.e., to the default
+        head location).
+    regularize : str | None
+        Basis regularization type, must be "in", "svd" or None.
+        "in" is the same algorithm as the "-regularize in" option in
+        MaxFilter™. "svd" (new in v0.13) uses SVD-based regularization by
+        cutting off singular values of the basis matrix below the minimum
+        detectability threshold of an ideal head position (usually near
+        the device origin).
+    ignore_ref : bool
+        If True, do not include reference channels in compensation. This
+        option should be True for KIT files, since Maxwell filtering
+        with reference channels is not currently supported.
+    """
+    if coord_frame not in ('head', 'meg'):
+        raise ValueError('coord_frame must be either "head" or "meg", not "%s"'
+                         % coord_frame)
+    origin = _check_origin(origin, info, 'head')
+    regularize = _check_regularize(regularize, ('in', 'svd'))
+    meg_picks, mag_picks, grad_picks, good_picks, coil_scale, mag_or_fine = \
+        _get_mf_picks(info, int_order, ext_order, ignore_ref)
+    info_from = pick_info(info, good_picks, copy=True)
+    all_coils_recon = _prep_mf_coils(info, ignore_ref=ignore_ref)
+    all_coils = _prep_mf_coils(info_from, ignore_ref=ignore_ref)
+    # remove MEG bads in "to" info
+    n_in, n_out = _get_n_moments([int_order, ext_order])
```

Comment:
Should this be modifying `exp`? `n_in` & `n_out` are not used and I don't see how bad channels are removed in the "to" `info`